### ISSUE TITLE
Next.js Build bugs 

### ISSUE THREAD

**Issue by loganpowell** [18F/federalist#2322]
#### User Story

As a federalist user, I want to be able to deploy any site that I could to github pages to federalist

#### Background (Optional)

I'm using Next.js and the routing is breaking for some reason when choosing "static html" option in build settings of fedralist

#### Acceptance Criteria

- [ ] go to [link](https://cg-8e6dfed8-1ef5-4632-a756-6c0407d417a7.app.cloud.gov/site/loganpowell/luca/)
- [ ] click on any link that takes you out of the page to either a static pdf or another size
- [ ] click the browser back button
- [ ] try not to get a 404 error

_After evaluating, edit this part:_

#### Implementation outline (if higher than "low" effort):

- [ ] <TODO>

**Comment by loganpowell** [comment]
Guys I found another issue where others are having this same experience:

https://github.com/zeit/next.js/issues/3065#issuecomment-521559286

I'm closing this in favor of that.

**Comment by loganpowell** [comment]
If anyone runs into this issue in the future, I found a not-so-pretty solution.

I'm not sure why `next export` doesn't work in `federalist` (or the underlying `s3` config), but does work in github pages. However, if you are using federalist with next export and you're running into `404` pages when either refreshing any page or navigating back from an external page to your next.js static site, I've found the following works:

If you add `.html` to your `as=` prop in any component using `next/link` for example:

Diff
```diff
import { withRouter } from 'next/router'
import Link from 'next/link'

const custom_link = ({ href, router, children, name, as }) => 
-   <Link prefetch href={href} as={as} passHref>
+   <Link prefetch href={href} as={as + ".html"} passHref>
      {children}
    </Link>

export const CustomLink = withRouter(ELink)

```
This works as it forces the server to serve the html file for the page instead of referring to the router which has lost its context... from what I gather. 

Hope this helps someone who's in a bind and needs a quick fix. Not an ideal solution, but eh...
